### PR TITLE
Handle clicking outside of a component and hide navigation on logout screen

### LINKED ISSUE #74: Close sidebar when clicking on the rest of the screen
## Description

The sidebar should be closed when the rest of the screen (not the sidebar) is clicked, instead of just the close button.

## Todos

These points are a rough guideline. **Please** feel free to discuss with others on the team about the best way to design this component!

- [x] Create a hook to detect outside clicks on an element, you'll want to do something similar to [this example](https://codesandbox.io/s/outside-alerter-hooks-lmr2y?module=/src/OutsideAlerter.js&file=/src/OutsideAlerter.js), but change the event from `mousedown` to `click` and instead of calling `alert(...)`, instead call a function that is passed into the hook (called `handler` or something, to handle the click outside)
- [ ] Use this hook in the sidebar component with a ref of the sidebar, and close the sidebar on an outside click.

## Useful links

- [Example outside clicker](https://codesandbox.io/s/outside-alerter-hooks-lmr2y?module=/src/OutsideAlerter.js&file=/src/OutsideAlerter.js)

### PR DESCRIPTION
**Pull request by ashayp22**
## Description

- Create `useClickedOutside()` hook for closing the sidebar when clicking outside of the sidebar.
- Move all custom hooks to `util/hooks`.
- Hide the `Sidebar` and `BottomNavBar` on the logout screen.

Fixes #74

### CODE REVIEW

File: lib/useClickedOutside.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useEffect, RefObject } from 'react';
+
+const useClickededOutside = (ref: RefObject<any>, handler: () => void) => {
```
Comment (AndrewLester) [inline lib/useClickedOutside.ts]: Typo "clicked". Fix other occurances
```suggestion
const useClickedOutside = (ref: RefObject<any>, handler: () => void) => {
```

**Review by AndrewLester — APPROVED** [review]
Small fix

File: lib/useClickedOutside.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { useEffect, RefObject } from 'react';
```
Comment (ericcccsliu) [inline lib/useClickedOutside.ts]: maybe we could make a `hooks` folder to put this and other react hooks in?

**Review by ericcccsliu — COMMENTED** [review]
LGTM

**Review by ericcccsliu — APPROVED** [review]
LGTM

**Review by AndrewLester — APPROVED** [review]
Thanks

**Pull request merged**